### ISSUE TITLE
Inconsistent/Confusing behavior of util "safe_div"

### ISSUE THREAD

**Issue by aobo-y** [pytorch/captum#654]
https://github.com/pytorch/captum/blob/master/captum/_utils/common.py#L26-L37

The behavior of how `safe_div` uses `default_value` is inconsistent depending on if `denorm` is `float` or `tensor`

Described in the the comments, seems the `default_value` should be used directly as the devision result when `denorm` is `0`. This is indeed implemented  accordingly when `denorm` is `float`
```py
return (numerator / denom) if denom != 0.0 else default_value
```

However, when `denorm` is `tensor`, the `default_value` is actually used as the `default denorm` instead
```py
return numerator / torch.where(denom != 0.0, denom, default_value)
```

I think the 2nd behavior is more reasonable, i.e., when `denorm` is `float`, update to
```py
return numerator / (denom if denom != 0.0 else default_value)
```

Also better rename the parameter to `default_denorm` to avoid further confusion.
The question left is if there is any need or use cases for the 1st bahvior.

**Comment by NarineK** [comment]
@aobo-y, that's a good point! I agree, let's change it to `return numerator / (denom if denom != 0.0 else default_value)`.
`default_denorm` is a good name.
Thank you!

**Comment by aobo-y** [comment]
fixed in #751
